src/thunderbots/software/test/primitive/primitive.cpp

```diff
@@ -72,6 +74,57 @@ TEST(PrimitiveTest, convert_MovePrimitive_to_message_and_back_to_MovePrimitive)
     EXPECT_EQ(new_prim->getExtraBits(), std::vector<bool>());
 }
 
+TEST(PrimitiveTest, creat_MoveSpinPrimitive_from_message_test)
+{
+    const Point destination     = Point(-1, 4);
+    const Angle angular_vel     = AngularVelocity::ofRadians(0.54);
+    const unsigned int robot_id = 2U;
+
+    MoveSpinPrimitive movespin_prim =
+        MoveSpinPrimitive(robot_id, destination, angular_vel);
+
+    thunderbots_msgs::Primitive prim_message = movespin_prim.createMsg();
+
+    MoveSpinPrimitive new_prim = MoveSpinPrimitive(prim_message);
+
+    std::vector<double> parameters = new_prim.getParameters();
+
+    EXPECT_EQ("MoveSpin Primitive", new_prim.getPrimitiveName());
+    EXPECT_EQ(robot_id, new_prim.getRobotId());
+    EXPECT_DOUBLE_EQ(destination.x(), parameters[0]);
+    EXPECT_DOUBLE_EQ(destination.y(), parameters[1]);
+    EXPECT_DOUBLE_EQ(angular_vel.toRadians(), parameters[2]);
+    EXPECT_EQ(movespin_prim.getExtraBits(), std::vector<bool>());
+}
+
+TEST(PrimitiveTest, creat_DirectWheelsPrimitive_from_message_test)
+{
+    const signed int wheel0_power = 2;
+    const signed int wheel1_power = 4;
+    const signed int wheel2_power = 6;
+    const signed int wheel3_power = 8;
+    const double dribbler_rpm     = 49.6;
+    const unsigned int robot_id   = 3U;
+
+    DirectWheelsPrimitive directwheels_prim = DirectWheelsPrimitive(
+        robot_id, wheel0_power, wheel1_power, wheel2_power, wheel3_power, dribbler_rpm);
+
+    thunderbots_msgs::Primitive prim_message = directwheels_prim.createMsg();
+
+    DirectWheelsPrimitive new_prim = DirectWheelsPrimitive(prim_message);
```

Comment:
```suggestion
    DirectWheelsPrimitive new_prim = Primitive::createPrimitive(prim_message);
```

We want to use this generic constructor because it will use/test the new `if/else` statements added in `primitive.cpp`